src/source/avro.rs

```diff
@@ -55,7 +88,23 @@ impl AvroStreamHandler {
             Err(_) => return,
         };
 
-        let mut buf = Vec::with_capacity(payload_size_in_bytes);
+        let header_size_in_bytes = mem::size_of::<Header>();
+        if payload_size_in_bytes <= header_size_in_bytes {
+            warn!("Invalid payload detected!  Discarding!");
```

Comment:
Worth adding self-telemetry for this case. People – maybe us – will ask for it.

Reply:
I agree 💯 on having telemetry here. I'm torn on what else we should do in this case. If this happens, the wire is probably fried. Currently this just reads the length N, but doesn't then toss out/seek past the next N bytes as invalid. In the next pass, we're going to read part or all of those N bytes as the next length, and potentially have an offset issue.

What's the right thing to do then? Toss out the N bytes as encoded in the length? Disconnect the socket? Do what we do now and just keep reading from the same position?

Reply:
Also: some sort of maximum payload size would be good to enforce. It's currently trivial to blow up the memory of the cernan process by sending a large length here.

Reply:
That'd be a good auditing project. To date we've assumed that cernan runs in a more or less trusted environment which, uh, is not even true for the environments it runs in today. One group goofing will cause a denial of service by exploding memory etc etc.